jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLClasspathHelperImpl.java

```diff
@@ -57,6 +57,8 @@
 		initialize(loader, id, canFind, canStore);
 		initializeShareableClassloader(loader);
 		initializeURLs();
+		notifyClasspathChange3(id, loader, this.urls, this.urlCount, true);
```

Comment:
new URLs are also added in addClasspathEntry(), which calls notifyClasspathChange2(), and this code path also needs to deal with setting the zip flags.

Reply:
also in setClasspath()

Reply:
Already add code to call `sharedClassConfig->updateClasspathOpenState()` in `notifyClasspathChange2()` to cover the case of `addClasspathEntry()` and `setClasspath()` as suggested.